### ISSUE TITLE
coreos-installer-growfs.service fails to start Grow root filesystem in Onlogic HX401 bare-metal hardware

### ISSUE THREAD

**Issue by mcattamoredhat** [coreos/coreos-installer-dracut#35]
After generating an UEFI bootable disk with osbuild-composer edge-simplified-installer, the os installation fails (one time failure).

After coreos-installer-service disk reading finishes, the service coreos-installer-growfs.service shows the following failure message:

```
[FAILED] Failed to start Grow root filesystem.
See 'systemctl status coreos-installer-growfs.service' for details.
[DEPEND] Dependency failed for Initrd Default Target.
```

This seems to be a one time failure, that means, after rebooting the system, the failure message does not appear again and the installation seems to be ok.

To generate UEFI bootable disk (edge-simplified-installer)
osbuild-composer-87-1.el9.x86_64
osbuild-91-1.el9.noarch
weldr-client-35.9-1.el9.x86_64

Bare Metal Hardware details:
Onlogic HX401

Steps to Reproduce:
1. Deploy RHEL-9.3 Openstack vm 
2. git clone https://github.com/virt-s1/rhel-edge.git
3. Edit simplified blueprint to specify proper installation device
$ vim ostree-simplified-installer.sh
  [customizations]
  - installation_device = "/dev/vda"
  + installation_device = "/dev/nvme0n1"
4. Generate iso UEFI bootable disk 
$ ./ostree-simplified-installer.sh
5. Write iso to bootable usb device
$ sudo dd if=123456-simplified-installer.iso of=/dev/sdx bs=1M status=progress
6. Boot bare-metal hardware from usb device

The experiment was repeat on another bare-metal hardware (Onlogic Karbon K410), that had /dev/sda as installation device name. The boot process and installation was successful.
This experiment is part of the verification of bug https://bugzilla.redhat.com/show_bug.cgi?id=2177645   

Filed BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2232406

**Comment by mcattamoredhat** [comment]
The output of lsblk command is showing that the size of rootfs is 9.5G.

```
[admin@localhost ~]$ lsblk
NAME                                          MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINTS
nvme0n1                                       259:0    0 119.2G  0 disk  
├─nvme0n1p1                                   259:1    0     1M  0 part  
├─nvme0n1p2                                   259:2    0   127M  0 part  /boot/efi
├─nvme0n1p3                                   259:3    0   384M  0 part  /boot
└─nvme0n1p4                                   259:4    0   9.5G  0 part
  └─luks-94da2594-8d33-4aba-b3d4-416a098861e6 253:0    0   9.5G  0 crypt
    └─rootvg-rootlv                           253:1    0     9G  0 lvm   /var
                                                                         /sysroot/ostree/deploy/redhat/var
                                                                         /usr
                                                                         /etc
                                                                         /
                                                                         /sysroot
```
Comparing that size with the command output of a successful installation generated by edge-installer, seems the partition above should be much bigger. Just like @perobins mentioned.

```
[admin@localhost ~]$ lsblk
NAME          MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINTS
nvme0n1       259:0    0 119.2G  0 disk  
├─nvme0n1p1   259:1    0   600M  0 part  /boot/efi
├─nvme0n1p2   259:2    0     1G  0 part  /boot
└─nvme0n1p3   259:3    0 117.7G  0 part  
  ├─rhel-root 253:0    0    70G  0 lvm   /var
  |                                      /usr
  |                                      /
  |                                      /sysroot
  ├─rhel-swap 253:1    0   7.6G  0 lvm   [SWAP]
  └─rhel-home 253:2    0  40.1G  0 lvm   /var/home

```

**Comment by nullr0ute** [comment]
So with a quick look I suspect we have to update the regex here:
https://github.com/coreos/coreos-installer-dracut/blob/main/dracut/scripts/coreos-installer-growfs#L29

**Comment by 7flying** [comment]
Fixed by #36
